elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/QueryBuilders.java

```diff
@@ -404,6 +408,52 @@ RangeQueryBuilder format(String format) {
     }
   }
 
+  /**
+   * A Query that does fuzzy matching for a specific value.
+   * It is equivalent to a wildcard search in Elasticsearch.
+   *
+   * <p>Attention:
+   * In Elasticsearch, the wildcard searches for the contents of the inverted index table,
+   * so the behavior is different for fields of type text and field type of keyword:
+   * 1.If the field type is keyword, es will index the entire field content,
+   * so the es wildcard search behavior is the same as sql's like.
+   * 2.If the field type is text, the content of the field will be saved into the index
+   * table after being segmented. So in this case, the es wildcard search is different
+   * from the like search in sql. In fact, at this time we recommend using full-text search
+   * instead of wildcard search.
+   *
+   */
+  static class WildcardQueryBuilder extends QueryBuilder {
+
+    private final String fieldName;
+    private final String value;
+
+    WildcardQueryBuilder(String fieldName, String value) {
+      this.fieldName = fieldName;
+      this.value = transformValue(value);
```

Comment:
Can `value` transformation happen earlier (like in `PredicateAnalyzer`)?
ES query builders shouldn't transform any values.